### PR TITLE
[web] Fixing text foreground paint / stroke for HTML web-renderer

### PR DESCRIPTION
**Pull request by abelokon0711**
This PR adds missing text stroke support when using the DOM and bitmap canvas by evaluating the TextStyle 'foreground' property.

On the DOM canvas it creates the corresponding CSS text stroke style. On the bitmap canvas it uses the HTML canvas strokeText method.

Fixes [#46683](https://github.com/flutter/flutter/issues/46683).

Is there a particular reason why in the current implementation the 'foreground' property is not used for painting text strokes in the HTML renderer?

Goldens: https://github.com/flutter/goldens/pull/230

### CODE REVIEW

**Comment by abelokon0711** [comment]
@googlebot I fixed it

File: lib/web_ui/lib/src/engine/text/paragraph.dart
Comment (mdebbar) [inline lib/web_ui/lib/src/engine/text/paragraph.dart:705]: Same here. Check for `foreground.style == ui.PaintingStyle.stroke`.

File: lib/web_ui/test/html/paragraph/general_golden_test.dart
Comment (mdebbar) [inline lib/web_ui/test/html/paragraph/general_golden_test.dart:535]: We need to put these screenshots in [flutter/goldens](https://github.com/flutter/goldens). I can do that once the PR is approved and ready to merge.

**Review by mdebbar — COMMENTED** [review]
Thanks for implementing this! I think this PR is almost ready to go after some minor suggestions are fixed.

File: lib/web_ui/lib/src/engine/html/bitmap_canvas.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/html/bitmap_canvas.dart]: Done.

File: lib/web_ui/lib/src/engine/html/bitmap_canvas.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/html/bitmap_canvas.dart]: Done.

File: lib/web_ui/lib/src/engine/html/bitmap_canvas.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/html/bitmap_canvas.dart]: Done.

File: lib/web_ui/lib/src/engine/text/paragraph.dart
Comment (abelokon0711) [inline lib/web_ui/lib/src/engine/text/paragraph.dart:699]: When comparing the outputs of the Bitmap Canvas and the DOM implementation, we have found, that we need to set the background color of the text to transparent to achieve the same effect as in the Bitmap Canvas and the Skia Engine that only the text stroke is painted.

File: lib/web_ui/lib/src/engine/text/paragraph.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: We fixed it, please check our comment for further information.

File: lib/web_ui/lib/src/engine/text/paint_service.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/text/paint_service.dart]: Done.

File: lib/web_ui/test/html/paragraph/general_golden_test.dart
Comment (abelokon0711) [reply lib/web_ui/test/html/paragraph/general_golden_test.dart:542]: That would be great, thanks!

**Review by mdebbar — COMMENTED** [review]
Thanks for the changes! This PR looks good to me with only 2 minor suggestions.

Once the [goldens PR](https://github.com/flutter/goldens/pull/230) is merged, you should update the [goldens_lock.yaml](https://github.com/flutter/engine/blob/main/lib/web_ui/dev/goldens_lock.yaml#L2) file to point to the new SHA in the flutter/goldens repo.

File: lib/web_ui/lib/src/engine/text/paragraph.dart
Comment (abelokon0711) [reply lib/web_ui/lib/src/engine/text/paragraph.dart]: Done.

**Comment by mdebbar** [comment]
@abelokon0711 I'm unable to add new commits to this PR. Could you please make sure that you checked the "Allow edits by maintainers" box?

**Comment by abelokon0711** [comment]
@mdebbar It's checked now 👍  Sorry for the inconvenience

**Review by mdebbar — APPROVED** [review]
Thanks for this great contribution!

**Comment by flar** [comment]
This PR is breaking the builds due to the undefined name "domRenderer".

See https://github.com/flutter/engine/pull/30038 for a potential fix.

**Pull request merged**
